File: app/src/main/java/com/kota/Telnet/TelnetAnsi.java
package com.kota.Telnet;

public class TelnetAnsi {
    public static byte DEFAULT_BACKGROUND_COLOR = 0;
    private static boolean DEFAULT_TEXT_BLINK = false;
    private static boolean DEFAULT_TEXT_BRIGHT = false;
    public static byte DEFAULT_TEXT_COLOR = 7;
    private static boolean DEFAULT_TEXT_ITALIC = false;
    public byte backgroundColor = 0;
    public boolean textBlink = false;
    public boolean textBright = false;
    public byte textColor = 0;
    public boolean textItalic = false;

    public TelnetAnsi() {
        resetToDefaultState();
    }

    public void resetToDefaultState() {
        this.textColor = DEFAULT_TEXT_COLOR;
        this.textBlink = DEFAULT_TEXT_BLINK;
        this.textBright = DEFAULT_TEXT_BRIGHT;
        this.textItalic = DEFAULT_TEXT_ITALIC;
        this.backgroundColor = DEFAULT_BACKGROUND_COLOR;
    }

    public static byte getDefaultTextColor() {
        return DEFAULT_TEXT_COLOR;
    }

    public static byte getDefaultBackgroundColor() {
        return DEFAULT_BACKGROUND_COLOR;
    }

    public static boolean getDefaultTextBlink() {
        return DEFAULT_TEXT_BLINK;
    }

    public static boolean getDefaultTextItalic() {
        return DEFAULT_TEXT_ITALIC;
    }
}

File: app/src/main/java/com/kota/telnet/TelnetAnsi.kt
package com.kota.telnet

class TelnetAnsi {
    var backgroundColor: Byte = 0
    var textBlink: Boolean = false
    var textBright: Boolean = false
    var textColor: Byte = 0
    var textItalic: Boolean = false

    init {
        resetToDefaultState()
    }

    fun resetToDefaultState() {
        this.textColor = DEFAULT_TEXT_COLOR
        this.textBlink = DEFAULT_TEXT_BLINK
        this.textBright = DEFAULT_TEXT_BRIGHT
        this.textItalic = DEFAULT_TEXT_ITALIC
        this.backgroundColor = DEFAULT_BACKGROUND_COLOR
    }

    companion object {
        const val DEFAULT_BACKGROUND_COLOR: Byte = 0
        const val DEFAULT_TEXT_BLINK: Boolean = false
        const val DEFAULT_TEXT_BRIGHT = false
        const val DEFAULT_TEXT_COLOR: Byte = 7
        const val DEFAULT_TEXT_ITALIC: Boolean = false
    }
}
